lib/elixir/lib/macro/env.ex

```diff
@@ -146,6 +155,89 @@ defmodule Macro.Env do
     [file: file, line: line]
   end
 
+  @doc """
+  Returns the alias for the given atom, `nil` otherwise.
+
+  ## Examples
+
+      iex> alias Foo.Bar, as: Baz
+      iex> Baz
+      Foo.Bar
+      iex> Macro.Env.alias_for(__ENV__, :Baz)
+      {:ok, Foo.Bar}
+      iex> Macro.Env.alias_for(__ENV__, :Unknown)
+      :error
```

Comment:
Why is the second argument an atom instead of an alias?

Reply:
Because:

```elixir
iex(3)> quote do Foo.Bar.Baz end
{:__aliases__, [alias: false], [:Foo, :Bar, :Baz]}
```

My goal is to be able to make aliases, imports, and requires more efficient in the future and the most efficient approach would be to not lookup an alias in the first place. :)